packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,288 @@
+import { Notifier, Observable, observable } from '@microsoft/fast-element';
+import {
+    DesignSystem,
+    ListboxElement as FoundationListbox,
+    ListboxOption
+} from '@microsoft/fast-foundation';
+import { keyEnter, keyEscape, keyTab } from '@microsoft/fast-web-utilities';
+import type { MentionDetail } from '../editor/types';
+import { styles } from './styles';
+import { template } from './template';
+import type { AnchoredRegion } from '../../anchored-region';
+import { diacriticInsensitiveStringNormalizer } from '../../utilities/models/string-normalizers';
+import type { ListOption } from '../../list-option';
+import type { MentionListBoxShowOptions } from './types';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-rich-text-mention-list-box': RichTextMentionListBox;
+    }
+}
+
+/**
+ * A rich text mention list box which acts as a popup for "@mention" support in editor
+ */
+export class RichTextMentionListBox extends FoundationListbox {
```

Comment:
The approach to derive from `FoundationListbox` here is nice because it is more consistent with how the select and combobox work. But it leaves us with a testing gap since we're no longer using the Nimble listbox directly. That means nothing is testing the visual appearance created by this component's styles and template. I think we need a storybook matrix test for this component. You should be able to copy the cases from the listbox test: cover all themes plus ensuring that long text renders with an ellipsis. If you run into problems with having multiple mention-list-box components open on the same page, you can copy patterns from menu button/select/combobox where we create separate stories for each theme. (You don't need separate stories for open above/below like those components have)

I was also expecting to see unit tests for this component. We at least need the basic test that verifies its tag is registered and that the component can be created. I'm still looking through the coverage in other files: is the remaining behavior (key handling, open/close) covered by integration tests in those components?

Reply:
Added matrix test and unit test for mention listbox. Key handling and open/close logic are tested under editor cases.